specifications/drtm-handler-and-table.rst

```diff
@@ -0,0 +1,310 @@
+
+2 - UEFI Environmnents
+2.1 - The DRTM Table shall be registed in the UEFI SystemTable
+2.1.2 - The protocol handle to be used is
+        '631c56b0-1780-4709-a450-8eec5f23e0c7'
+2.2 - On construction of the DRTM Table, the EFI Info record will be added
+2.3 - All EFI bootloaders shall record an EFI Config record with EFI Config
+      Entry for each configuration action on or information provide to the DLME
+      kernel
+2.4 - The bootloader responsible for calling EndBootServices() shall be
+      responsible for calling DLE Handler
+2.4.1 - The address for the DRTM Table will passedin the architecture specific register
+
+3 - x86 Platforms
+3.1 - On Intel TXT platforms the location of the DRTM Table will be stored in
+      the OS2MLE structure, see Appendix A
+3.2 - On AMD platforms the location of the DRTM Table will be stored in the
+      Secure Kernel Loader (SKL) configuration table
+
+4 - Arm Platforms
+4.1 - TBD
+
+Structure
+^^^^^^^^^
+
+The table is constructed from a Tag-Length-Value list consisting of elements
+described in the following sections. For general portability, the definitions
+for each element are presented as C structures.
+
+
+DRTM Table
+""""""""""
+
+.. code-block::
+   :linenos:
+
+    struct drtm_table {
+        uint32_t magic; /* 0x4452544d */
+        uint16_t revision;
+        uint16_t architecture;
+        uint32_t table_size;
+        uint32_t table_max_size
+        struct drtm_entry_hdr entries[0];
+    };
+
+DRTM Table Entry
+""""""""""""""""
+
+.. code-block::
+   :linenos:
+
+    struct drtm_entry_hdr {
+        uint16_t tag;
+        uint16_t size;
+    };
+
+
+DRTM Dynamic Launch Configuration
+"""""""""""""""""""""""""""""""""
+
+.. code-block::
+   :linenos:
+
+    /* TAG: 0x0008 */
+    struct drtm_entry_dl_info {
+        struct drtm_entry_hdr hdr;
+        uint64_t dl_handler;
+        uint64_t dce_base;
+        uint32_t dce_size;
+        uint64_t dlme_entry;
+    };
+
+DRTM Log
+""""""""
+
+.. code-block::
+   :linenos:
+
+    /* TAG: 0x0009 */
+    struct drtm_entry_log {
+        struct drtm_entry_hdr hdr;
+        uint16_t format;
+        uint16_t reserved;
+        uint64_t addr;
+        uint32_t size;
+    };
+
+
+DRTM Policy
+"""""""""""
+
+.. code-block::
+   :linenos:
+
+    /* TAG: 0x000a */
+    struct drtm_entry_policy {
+        struct drtm_entry_hdr hdr;
+        uint16_t revision;
+        uint16_t reserved;
+        uint16_t algorithms
+        uint16_t nr_entries;
+        struct policy_entry entries[0];
+    };
+  
+
+DRTM Policy Entry
+"""""""""""""""""
+
+A policy entry represents an entity that the SL Entry is being requested to
+measure. As an SL Entry is able to measure an attribute of a kernel, that
+attribute will be published as an entity type. A general "memory location"
+entity type is available for measuring a range of memory.
+
+.. code-block::
+   :linenos:
+
+    struct drtm_policy_entry {
+        uint16_t pcr;
```

Comment:
A policy must be well formed and all mandatory entries must be there. The code should check to ensure that the PCR value is in fact a DRTM PCR. This also make CONFIG_SECURE_LAUNCH_ALT_DLME_AUTHORITY and CONFIG_SECURE_LAUNCH_ALT_DLME_DETAIL obsolete.